subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -1930,27 +1949,54 @@ static int ztls_poll_update_ctx(struct tls_context *ctx,
 	const struct fd_op_vtable *vtable;
 	void *obj;
 	int ret;
+	short events = pfd->events;
 
 	obj = z_get_fd_obj_and_vtable(
 		ctx->sock, (const struct fd_op_vtable **)&vtable);
 	if (obj == NULL) {
 		return -EBADF;
 	}
 
+	/* Check if the socket was waiting for the handshake to complete. */
+	if ((pfd->events & ZSOCK_POLLIN) &&
+	    ((*pev)->obj == &ctx->tls_established)) {
+		/* In case handshake is complete, reconfigure the k_poll_event
+		 * to monitor the underlying socket now.
+		 */
+		if ((*pev)->state != K_POLL_STATE_NOT_READY) {
+			ret = z_fdtable_call_ioctl(vtable, obj,
+						   ZFD_IOCTL_POLL_PREPARE,
+						   pfd, pev, *pev + 1);
+			if (ret != 0 && ret != -EALREADY) {
+				return ret;
+			}
+
+			return -EAGAIN;
```

Comment:
Re: discussion above: I see, but please add comment here along the lines of: "signal to poll() that it should make (if needed) another iteration with the event reconfigured above".

Reply:
Sure, I've added the comment.